weave-js/src/components/Button/Button.tsx

```diff
@@ -4,163 +4,172 @@
-              'text-moon-800 dark:text-moon-200': isSecondary || isGhost,
-              'hover:bg-teal-300/[0.48] hover:text-teal-600 dark:hover:bg-teal-700/[0.48] dark:hover:text-teal-400':
-                isSecondary || isGhost,
-              'bg-teal-300/[0.48] text-teal-600 dark:bg-teal-700/[0.48] dark:text-teal-400':
-                (isSecondary || isGhost) && active,
-
-              // quiet
-              'text-moon-500': isQuiet,
-              'bg-oblivion/[0.05] text-moon-800 dark:bg-moonbeam/[0.05] dark:text-moon-200':
-                isQuiet && active,
-              'hover:text-moon-800 dark:hover:text-moon-200': isQuiet,
-              'hover:bg-oblivion/[0.05] dark:hover:bg-moonbeam/[0.05]': isQuiet,
-
-              // destructive
-              'bg-red-500 text-white hover:bg-red-450': isDestructive,
-              'bg-red-450': isDestructive && active,
-            },
-            className
-          )
-        )}
-        {...htmlAttributes}>
-        {startIcon ? <Icon name={startIcon} /> : null}
-        {children}
-        {endIcon ? <Icon name={endIcon} /> : null}
-      </button>
+    const classes = twMerge(
+      classNames(
+        // Display
+        'inline-flex items-center justify-center',
+        // Font
+        "font-['Source_Sans_Pro'] font-semibold",
+        // Disabled
+        'disabled:pointer-events-none disabled:opacity-35',
+        // Focused
+        'focus-visible:outline focus-visible:outline-[2px] focus-visible:outline-teal-500',
+        // Other
+        'night-aware whitespace-nowrap rounded border-none',
+        sizeClasses[size],
+        variantClasses[variant],
+        // Conditional classes
+        {
+          [iconOnlySizeClasses[size]]: Boolean(icon),
+          [activeVariantClasses[variant]]: active,
+        },
+        className
+      )
     );
 
-    const [isTooltipOpen, setIsTooltipOpen] = useState(false);
-    if (tooltip) {
-      return (
-        <Tailwind style={wrapperStyles}>
-          <Tooltip.Provider>
-            <Tooltip.Root open={isTooltipOpen} onOpenChange={setIsTooltipOpen}>
-              <Tooltip.Trigger asChild>
-                {/* span is needed so tooltip works on disabled buttons */}
-                <span>{button}</span>
-              </Tooltip.Trigger>
-              <Tooltip.Portal>
-                <Tooltip.Content {...tooltipProps}>{tooltip}</Tooltip.Content>
-              </Tooltip.Portal>
-            </Tooltip.Root>
-          </Tooltip.Provider>
-        </Tailwind>
-      );
-    }
-
-    return <Tailwind style={wrapperStyles}>{button}</Tailwind>;
+    return (
+      <Tailwind
+        /**
+         * The Tailwind wrapper is a div, which is a block-element. However, a normal button
+         * wouldn't be wrapped and would be inline-block by default. These styles are a
+         * necessary workaround to ensure Button is styled correctly despite the wrapper.
+         */
+        style={{display: 'contents'}}>
```

Comment:
hmm it would be nice if this just worked, but i did some quick testing in our storybook and it doesn't behave the same as before

before:
<img width="811" alt="Screenshot 2024-08-20 at 2 30 31 PM" src="https://github.com/user-attachments/assets/fec01030-3cef-4db4-9813-4e3aeed68a93">

after:
<img width="818" alt="Screenshot 2024-08-20 at 2 30 11 PM" src="https://github.com/user-attachments/assets/d39eda5d-9c97-4965-8089-d2457ac8a8b6">

Reply:
oh yikes, thanks for checking 😅 

IMO the change here to make it align with the previous version should be implemented as a class rather than overriding things in the tailwind wrapper. I also think exposing the tailwind props is fraught with peril. I'll take a look at what I can do to fix this though 👍🏼